src/components/Battle.js

```diff
@@ -89,48 +100,66 @@ class Battle extends Component {
       moves: state.maxMoves,
       handArray: newHandArray,
       deckArray: newdeckArray,
-      discardArray: newDiscardArray
+      discardArray: newDiscardArray,
+      actionQueue: []
     }));
   }
 
   render() {
     return (
       <div>
         <MonsterWrapper>
-          {this.props.monsters.map(monster => {
+          {this.props.monstersState.map(monster => {
             return <Monster key={monster.id} hp={monster.hp} />;
           })}
         </MonsterWrapper>
         <br />
         <BattleStats>
-          Player HP:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{this.props.player.hp}
+          Player HP:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{this.props.playerState.hp}
           <br />
           Deck Size:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
-          {this.props.cards.length}
+          {this.props.cardsState.length}
           <br />
-          Cards in Hand:&nbsp;&nbsp;{this.props.player.hand.length}
+          Cards in Hand:&nbsp;&nbsp;{this.props.playerState.hand.length}
           <br />
-          Discard Pile:&nbsp;&nbsp;&nbsp;{this.props.player.discard.length}
+          Discard Pile:&nbsp;&nbsp;&nbsp;{this.props.playerState.discard.length}
           <br />
           Actions:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
-          {this.props.player.remainingActions}/{this.props.player.maxActions}
+          {this.props.playerState.remainingActions}/{this.props.playerState.maxActions}
           <br />
         </BattleStats>
         <br />
         <Hand>
-          {this.props.player.hand
-            .map(id => this.props.cards.find(card => card.id === id))
+          { this.props.playerState.hand
+            .map(id => this.props.cardsState.find(card => card.id === id))
             .map(cardInHand => {
-              let card = new Card({
-                label: cardInHand.name,
-                value: cardInHand.power,
-                key: cardInHand.id,
-                cost: cardInHand.cost,
-                description: cardInHand.description
-              });
-
-              return card.render();
-            })}
+              switch(cardInHand.id) {
+                case 1:
+                case 2:
+                  return <Card label={cardInHand.name}
+                      value={cardInHand.power}
+                      key={cardInHand.id}
+                      uniqueId={cardInHand.id}
+                      cost={cardInHand.cost}
+                      actions={
+                        [
+                          monsters.attackMonster(1, cardInHand.power),
```

Comment:
Note that we can't use `this.props.attackMonster` here because we want to pass the action without triggering it and `this.props.attackMonster` is bound to `dispatch`, so we call the action creator directly.